spec/test/test_validate.py

```diff
@@ -0,0 +1,191 @@
+"""
```

Comment:
New file -- testing validation of the different types of specs.

Reply:
Very thorough!!